packages/apps/comment-nuke/src/nuke.ts

```diff
@@ -1,13 +1,128 @@
+    const startTime = new Date();
+    let success = true;
+    let message: string;
+  
+    const shouldLock = props.lock;
+    const shouldRemove = props.remove;
+    const skipDistinguished = props.skipDistinguished;
+  
+    if (!shouldLock && !shouldRemove) {
+      return { message: 'You must select either lock or remove.', success: false };
+    }
+  
+    try {
+      const post = await context.reddit.getPostById(props.postId);
+      const user = await context.reddit.getCurrentUser();
+      const modPermissions = await user.getModPermissionsForSubreddit(post.subredditName);
+      const canManagePosts = modPermissions.includes('all') || modPermissions.includes('posts');
+ 
+      console.log(
+        `Mod Info: r/${post.subredditName} u/${user.username} permissions:${modPermissions}: ${
+          canManagePosts ? 'Can mod' : 'Cannot mod'
+        }`
+      );
+  
+      if (!canManagePosts) {
+        console.info('A user without the correct mod permissions tried to nuke all comments of a post.');
+        return {
+          message: 'You do you not have the correct mod permissions to do this.',
+          success: false,
+        };
+      }
+  
+      const allComments = await getAllCommentsInPost(post);
+      const comments = allComments.filter(
+        (eachComment: any) => !(skipDistinguished && eachComment.isDistinguished())
+      );
+  
+      const lockPromises: Promise<any>[] = [];
+      const removePromises: Promise<any>[] = [];
+  
+      for (const eachComment of comments) {
+        const commentId = eachComment.id;
+        if (shouldLock && !eachComment.locked) {
+          lockPromises.push(eachComment.lock());
+        }
+  
+        if (shouldRemove && !eachComment.removed) {
+          removePromises.push(eachComment.remove());
+        }
+      }
+      const responses = await Promise.all([...removePromises, ...lockPromises]);
+  
+      for (const r of responses) {
+        if (r && r.status === 'rejected') {
+          console.error('Failed to remove or lock a comment.');
+          console.info(r.reason);
+        }
+      }
+  
+      const verbage =
+        shouldLock && shouldRemove ? 'removed and locked' : shouldLock ? 'locked' : 'removed';
+  
+      if (shouldRemove) {
+        const postId = props.postId;
+        await context.modLog
+          .add({
+            action: 'removecomment',
+            target: postId,
+            details: 'comment-mop app',
+            description: `u/${user.username} used comment-mop to ${verbage} all comments of this post.`,
+          })
+          .catch((e: any) =>
+            console.error(`Failed to add modlog for post: ${postId}.`, e.message)
+          );
+      }
+  
+      success = true;
+      message = `${comments.length} comment${
+        comments.length > 1 ? 's' : ''
+      } ${verbage}! Refresh the page to see the cleanup.`;
+  
+      const finishTime = new Date();
+      const timeElapsed = (finishTime as any) - (startTime as any);
```

Comment:
I think we can hold use Date.now() instead to get the toal number of milliseconds to avoid a cast down here.